### PR TITLE
Make cennznet use the latest plug and the fix for the transaction pool

### LINKED ISSUE #233: Bring the latest Plug changes to Cennznet
This is a prep step for another release of cennznet.

### PR DESCRIPTION
**Pull request by aliXsed**
closes #233

### CODE REVIEW

**Review by KarishmaBothara — APPROVED** [review]

File: cli/src/service.rs [hunk 1]
```diff
@@ -355,84 +359,212 @@ pub fn new_light(config: Configuration) -> Result<impl AbstractService, ServiceE
...
+	use sc_consensus_epochs::descendent_query;
+	use sc_service::AbstractService;
+	use sp_consensus::{
+		BlockImport, BlockImportParams, BlockOrigin, Environment, ForkChoiceStrategy, Proposer, RecordProof,
+	};
+	use sp_core::{crypto::Pair as CryptoPair, H256};
+	use sp_finality_tracker;
+	use sp_keyring::AccountKeyring;
+	use sp_runtime::traits::IdentifyAccount;
+	use sp_runtime::{
```
Comment (benxgao) [inline cli/src/service.rs:377]: Squash `traits::IdentifyAccount` inside here?

File: cli/src/service.rs [hunk 2]
```diff
@@ -355,84 +359,212 @@ pub fn new_light(config: Configuration) -> Result<impl AbstractService, ServiceE
...
+
 		sc_service_test::sync(
-			sc_chain_spec::integration_test_config(),
-			|config| new_full(config),
-			|mut config| new_light(config),
-			block_factory,
-			extrinsic_factory,
+			chain_spec,
+			|config| {
+				let mut setup_handles = None;
```
Comment (benxgao) [inline cli/src/service.rs:411]: Not sure if CI/CD will auto fix it, have you run `cargo fmt` locally?

File: cli/src/service.rs (hunk 1 above)
Comment (aliXsed) [reply cli/src/service.rs:377]: It's matching the corresponding file in Plug. We'd better not to make such changes in cennznet to keep it comparable with the source for future updates.

File: cli/src/service.rs (hunk 2 above)
Comment (aliXsed) [reply cli/src/service.rs:411]: Yes and it has passed the CI/CD formatting check

File: cli/src/service.rs (hunk 1 above)
Comment (benxgao) [reply cli/src/service.rs:377]: May need to check with tech leads on code standards to get the team on the same page. cc @holygits

File: cli/src/service.rs (hunk 2 above)
Comment (benxgao) [reply cli/src/service.rs:411]: Formatted code is not just for easier development but also for better review experiences. I surppose github code review formatting is based on intents we submitted. May need to check with tech leads if we considered coding standards for further commits? cc @holygits

**Comment by benxgao** [comment]
Got few concerns in coding standards, otherwise, would approve this PR for team internal use purpose, now that this is not an official release branch.

**Review by benxgao — APPROVED** [review]

**Pull request merged**